core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1302,9 +1311,68 @@ public boolean processO3Block(
         }
     }
 
-    public void processWalCommit(
-            Path walPath,
-            long segmentId,
+    private void closeWalColumns() {
+        for (int col = 0, n = walMappedColumns.size(); col < n; col++) {
+            MemoryCMOR mappedColumnMem = walMappedColumns.getQuick(col);
+            if (mappedColumnMem != null) {
+                Misc.free(mappedColumnMem);
+                walColumnMemoryPool.push(mappedColumnMem);
+            }
+        }
+        walEventReader.close();
+    }
+
+    public void processWalCommit(@Transient Path walPath, long segmentTxn, SqlToOperation sqlToOperation) {
+        final WalEventCursor walEventCursor = walEventReader.of(walPath, WAL_FORMAT_VERSION, segmentTxn);
```

Comment:
We seem to close `walEventReader` only in case of DATA here, as a part of the `closeWalColumns()` method, which doesn't seem nice. Better to have try/finally blocks here and make sure that we close it in this method, not in `closeWalColumns()`.

Reply:
makes sense, changed it